torch/nn/parallel/distributed.py

```diff
@@ -185,7 +185,7 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0,
                 if not param_tuple[0].requires_grad:
                     continue
                 for p in param_tuple:
-                    self.bucket_map[p] = bucket_idx
+                    self.bucket_map[id(p)] = bucket_idx
```

Comment:
Need a fix to `__setstate__`.

Reply:
I think one way to do this is to create a copy of the new state to be set, except that the keys of the copy would be the `id`s of the original new state, and then pass this to `__setstate__`. Does this sound okay?